File: app/src/main/java/com/legue/axel/lolsummonertool/database/converter/IntegerListConverters.java
package com.legue.axel.lolsummonertool.database.converter;

import android.arch.persistence.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.Collections;
import java.util.List;

public class IntegerListConverters {

    @TypeConverter
    public static List<Integer> integerToList(String value) {
        Gson gson = new Gson();

        if (value == null) {
            return Collections.emptyList();
        }

        Type listType = new TypeToken<List<Integer>>() {
        }.getType();

        return gson.fromJson(value, listType);
    }

    @TypeConverter
    public static String ListToString(List<Integer> descriptions) {
        Gson gson = new Gson();
        return gson.toJson(descriptions);
    }
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/converter/IntegerListConverters.kt
package com.legue.axel.lolsummonertool.database.converter

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class IntegerListConverters {

    @TypeConverter
    fun integerToList(value: String?): List<Int>? {
        if (value == null || value == "null") {
            return emptyList()
        }

        val listType = object : TypeToken<List<Int>>() {}.type
        val objects = Gson().fromJson(value, listType) as List<Int>
        return objects.toList()
    }

    @TypeConverter
    fun ListToString(descriptions: List<Int>): String {
        return Gson().toJson(descriptions)
    }
}
